### DISCUSSION TITLE
How to Implement Logical Isolation in EMQx?

### DISCUSSION THREAD

**Discussion by wangfeiing** [emqx/emqx#10787] [Q&A] [answered]
How can EMQX achieve logical isolation when multiple business product lines are used?
For example, A and B are different product lines, how can functional isolation between A and B be achieved?

**Comment by lafirest** [comment, marked as answer]
@wangfeiing  If you want to mix different product lines in one broker, usually, you should use different topic prefixes to distinguish them, e.g. the topic model could be: productId/raw-topic.
If you don't like prefixes, you could use different ports which has a mount point for each product, the mount point is transparent for clients. See this [listeners.tcp.$name.mountpoint](https://www.emqx.io/docs/en/v5.0/configuration/configuration-manual.html#mqtt-tcp-listener-1883) to know what is the mount point.

**Reply by wangfeiing** [reply]
Thank you for your reply. 
In addition, we are going to use a multi-cloud and multi-active architecture, how should the core nodes be deployed?
I'm going to do this. There are 3 core nodes and N replication nodes on one cloud, and 3 core nodes and N replication nodes on another cloud. These 6 + 2*N nodes together form a network  emqx cluster.

**Reply by lafirest** [reply]
The EMQX cluster expected a low-latency network environment, the connection between the core and replicant can tolerate higher latency,
Some feedback from our users said it is better to limit the core-to-core latency to 10ms, and the core-to-replicate limit to 30ms.
Maybe you need to set up a real test cluster to find the best way.
